### ISSUE TITLE
jsx-first-prop-new-line: option to ignore boolean props when value is omitted

### ISSUE THREAD

**Issue by ssalka** [yannickcr/eslint-plugin-react#2455]
Currently, this raises a warning (using [Material UI Button](https://material-ui.com/api/button) as an example):
```jsx
<Button dense
  className="my-btn"
  onClick={handleClick}
/>
```

But I would like for it not to - it seems wasteful to use a whole line just to put a single word, _especially when it is used as a modifier_, like it is here.

If there are several such modifier props, you can save a lot of space:

```jsx
<Button dense centerRipple fullWidth
  className="my-btn"
  onClick={handleClick}
/>

// vs...

<Button
  dense
  fullWidth
  centerRipple
  className="my-btn"
  onClick={handleClick}
/>
```

Obviously you wouldn't want to put too many props on one line, but with just a few I don't think it sacrifices any readability.

This style could be allowed through an option, e.g. `ignoreBoolModifiers`, which can default to `false`.

Note, though this would apply specifically to boolean props, the following should still be disallowed:
```jsx
<Button dense={true}
  className="my-btn"
  onClick={handleClick}
/>
```

**Comment by ljharb** [comment]
I think you've certainly described a straightforward option; I'm very much not convinced of the value though. "Saving space", like line count, isn't a good thing to strive for/focus on, but rather clarity and consistency.

**Comment by ssalka** [comment]
Thanks for your feedback - I feel that the co-location of the component (Button) and the modifier prop (dense) improves readability; saving lines is not the desired outcome but rather a nice side-effect, though now I see my last post didn't quite come off that way. An additional side-effect is that it can allow you to separate out some props that are _definitely static_ (modifier props are always true), while keeping the potentially dynamic props each on their own line.

Generally I'm a fan of configurability, and want users to have options to choose from when they are not satisfied with the defaults - in this case, I fall into that class of users 😅

Without said option, my preference is to turn this rule off, which accommodates for my prop usage, though it still allows the `dense={true}` variant, which I'll just have to be careful to avoid.

**Comment by ljharb** [comment]
`dense={true}` is also always static - always true.

**Comment by ljharb** [comment]
Since nobody else has asked for this in the last couple years, I'm going to close this.
